src/main/java/at/ac/fhcampuswien/snake/board/GameBoard.java

```diff
@@ -0,0 +1,518 @@
+        int randomY = getRandomWallPosition(rand, wallLength, false);
+
+        return new Wall(rand.nextBoolean(), randomX, randomY, wallLength);
+    }
+
+    /**
+     * This method generates a random coordinate for a wall on the board.
+     * It adds the snake's position to a Set of exclusions to avoid creating the wall on top of it
+     *
+     * @param rand         - the random generator
+     * @param wallLength   - the total wall length
+     * @param isHorizontal - the type of coordinate to generate
+     * @return a random int for the starting position of the wall
+     */
+    private int getRandomWallPosition(Random rand, int wallLength, boolean isHorizontal) {
+        int range = GAME_BOARD_SIZE_MEDIUM - OBJECT_SIZE_MEDIUM * (wallLength + 2);
+
+        Set<Integer> exclusions = new HashSet<>();
+        int segmentPosition;
+        for (Position segment : snake.getSegments()) {
+            if (isHorizontal) {
+                segmentPosition = segment.getX();
+            } else {
+                segmentPosition = segment.getY();
+            }
+            exclusions.add(segmentPosition);
+            for (int i = 0; i < wallLength; i++) {
+                exclusions.add(segmentPosition + i * OBJECT_SIZE_MEDIUM);
+                exclusions.add(segmentPosition - i * OBJECT_SIZE_MEDIUM);
+            }
+        }
+
+        exclusions.add(0);
+        exclusions.add(OBJECT_SIZE_MEDIUM);
+        exclusions.add(OBJECT_SIZE_MEDIUM * 2);
+
+        int random = rand.nextInt(range) / OBJECT_SIZE_MEDIUM * OBJECT_SIZE_MEDIUM;
+        while (exclusions.contains(random)) {
+            random = rand.nextInt(range) / OBJECT_SIZE_MEDIUM * OBJECT_SIZE_MEDIUM;
+        }
+        return random;
+    }
+
+    /**
+     * Method to draw the checkerboard pattern on the GraphicsContext
+     *
+     * @param gc GraphicsContext gc used for all BoardObjects
+     */
+    private void drawGameBoard(GraphicsContext gc) {
+        for (int i = 0; i < GAME_BOARD_SIZE_MEDIUM; i++) {
+            for (int j = 0; j < GAME_BOARD_SIZE_MEDIUM; j++) {
+                if ((i + j) % 2 == 0) {
+                    gc.setFill(Color.web(GAMEBOARD_COLOR_LIGHT));
+                } else {
+                    gc.setFill(Color.web(GAMEBOARD_COLOR_DARK));
+                }
+                gc.fillRect(i * OBJECT_SIZE_MEDIUM, j * OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+            }
+        }
+    }
+
+    private void drawFood(GraphicsContext gc, Food food) {
+        Image foodImg = new Image("graphics/food/" + food.getFoodType());
+        gc.drawImage(foodImg, food.getLocation().getX(), food.getLocation().getY(), OBJECT_SIZE_MEDIUM, OBJECT_SIZE_MEDIUM);
+    }
+
+    private void drawSnake(GraphicsContext gc) {
+        gc.setFill(Color.BLUE);
+        Position headPosition = snake.getSegments().get(0);
+        Image head = snakeHeadUp;
+
+        switch (snake.getDirection()) {
```

Comment:
Check this switch. Maybe we even do not need this? Please explain? Do we need this in this way?